docs: react native guide (#7166)
vuestorefront/vue-storefront: docs/content/guides/2.alokai-essentials/2.alokai-react-native/3.install-middleware.md

---
title: Install Alokai Middleware
layout: default
navigation:
  icon: tabler:number-2-small
---

# Install and Configure Alokai Middleware

In the last section, we have installed the React Native application and created a new Alokai project. In this section, we will install and configure the Alokai Middleware.

The Alokai Middleware is a Node.js application that acts as a bridge between the frontend application and the backend. It is responsible for handling all the API requests and responses, as well as managing the state of the application.

In this guide, we will install the Alokai Middleware and configure it to work with SAP Commerce Cloud.

## Install Alokai Middleware

Navigate to a newly generated `middleware` directory and install the project dependencies by running the following command:

```bash
cd storefront-middleware
yarn add @vue-storefront/middleware consola ts-node-dev dotenv typescript
```

This will install all the necessary dependencies for the `middleware` application.

![Installing VSF Middleware](./images/rn-starter-3.webp)

## Install Alokai SAP Commerce Cloud Integration

> So far, we have installed the `storefront-middleware` application. But in order for it to run and communicate with SAP Commerce Cloud, we need to install the `@vsf-enterprise/sapcc-api` package. This package is a Alokai integration for SAP Commerce Cloud.

Reviewer: ```suggestion
So far, we have installed the `storefront-middleware` application. But in order for it to run and communicate with SAP Commerce Cloud, we need to install the `@vsf-enterprise/sapcc-api` package. This package is an Alokai integration for SAP Commerce Cloud. 

It will add endpoints to your Middleware that will communicate directly with SAP. 
```

To do this, you will need to create a new configuration file in the `storefront-middleware` directory. Create a new file called `middleware.config.ts` and add the following code:

```typescript
require('dotenv').config();

export const integrations = {
  sapcc: {
    location: '@vsf-enterprise/sapcc-api/server',
    configuration: {
      OAuth: {
        uri: process.env.SAPCC_OAUTH_URI,
        clientId: process.env.SAPCC_OAUTH_CLIENT_ID,
        clientSecret: process.env.SAPCC_OAUTH_CLIENT_SECRET,
        tokenEndpoint: process.env.SAPCC_OAUTH_TOKEN_ENDPOINT,
        tokenRevokeEndpoint: process.env.SAPCC_OAUTH_TOKEN_REVOKE_ENDPOINT,
        cookieOptions: {
          'vsf-sap-token': { secure: process.env.NODE_ENV !== 'development' }
        }
      },
      api: {
        uri: process.env.SAPCC_API_URI,
        baseSiteId: process.env.DEFAULT_BASE_SITE_ID,
        catalogId: process.env.DEFAULT_CATALOG_ID,
        catalogVersion: process.env.DEFAULT_CATALOG_VERSION,
        defaultLanguage: process.env.DEFAULT_LANGUAGE,
        defaultCurrency: process.env.DEFAULT_CURRENCY
      }
    }
  }
};
```

> This configuration file will be used to configure the `middleware` application to work with SAP Commerce Cloud. It includes all the necessary configuration options for the SAP Commerce Cloud integration. 

Reviewer: ```suggestion
This configuration file will includes all of the necessary options for the `middleware` application to work with SAP Commerce Cloud.
```
